src/backend/gpopt/translate/CTranslatorDXLToPlStmt.cpp

```diff
@@ -4724,6 +4730,16 @@ CTranslatorDXLToPlStmt::TranslateDXLTblDescrToRangeTblEntry(
 
 	rte->eref = alias;
 
+	/*
+		Store the alias information only when table alias is used to allow 
+		EXPLAIN find it
+	*/
+	if (optimizer_enable_table_alias && NULL != table_descr->MdAlias())
+	{
+		rte->alias = (Alias *) copyObject(alias);
```

Comment:
Hoe about to move it to above same condition?

Reply:
Prefer to make a deep copy after filling fields than before. Copy is used here because in the similar places where both eref and alias initialized copy is used.

Reply:
Why do we need deep copy here?

Reply:
Same reason as for:
https://github.com/arenadata/gpdb/blob/2fd27bbc76ae4547cd99d1edac260805f6655cd6/src/backend/parser/analyze.c#L1095
https://github.com/arenadata/gpdb/blob/2fd27bbc76ae4547cd99d1edac260805f6655cd6/src/backend/parser/parse_relation.c#L1225
https://github.com/arenadata/gpdb/blob/2fd27bbc76ae4547cd99d1edac260805f6655cd6/src/backend/parser/parse_relation.c#L1678
https://github.com/arenadata/gpdb/blob/2fd27bbc76ae4547cd99d1edac260805f6655cd6/src/backend/parser/parse_relation.c#L1761
https://github.com/arenadata/gpdb/blob/2fd27bbc76ae4547cd99d1edac260805f6655cd6/src/backend/parser/parse_relation.c#L1850
Perhaps it's done to allow modify and use them independently in the code.

Reply:
Everything mentioned is C code, how does this apply to our case? Where are they changed and used independently in ORCA?

Reply:
RLE here is prepared for EXPLAIN which is in backend/command and is  not part of ORCA. It's not used in ORCA.

Reply:
> It's not used in ORCA

https://github.com/arenadata/gpdb/pull/780/files#diff-c66cca54786d2aba98ba3a9b644ce710180f227f699b3247f302d19a498dc69eR139-R145

Reply:
rte is used not only in the ORCA, but also in C-code at least in ExplainNode. I have feeling that it's much more safe to have a deep copy, similar to other places in C code.